core/src/main/java/bio/terra/pearl/core/service/publishing/PublishingUtils.java

```diff
@@ -57,4 +57,12 @@ public static <T extends BaseEntity & Versioned, D extends BaseVersionedJdbiDao<
             service.assignPublishedVersion(entity.getId());
         }
     }
+
+    public static void setPropertyEnumSafe(Object object, String propertyName, Object newValue) throws Exception {
+        if(object.getClass().getDeclaredField(propertyName).getType().isEnum()) {
```

Comment:
Nit: Totally a style thing, but this might read better with an assignment in the `if` clause and a single call to `setProperty`.